packages/server/graphql/public/typeDefs/OAuth.graphql

```diff
@@ -0,0 +1,113 @@
+extend type Organization {
+  oauthProviders: [OAuthAPIProvider!]!
+}
+
+extend type Mutation {
+  """
+  Create a new OAuth 2.0 Application
+  """
+  createOAuthAPIProvider(input: CreateOAuthAPIProviderInput!): CreateOAuthAPIProviderPayload!
+
+  """
+  Update an OAuth 2.0 Application
+  """
+  updateOAuthAPIProvider(input: UpdateOAuthAPIProviderInput!): UpdateOAuthAPIProviderPayload!
+
+  """
+  Delete an OAuth 2.0 Application
+  """
+  deleteOAuthAPIProvider(input: DeleteOAuthAPIProviderInput!): DeleteOAuthAPIProviderPayload!
+
+  """
+  Regenerate the OAuth 2.0 client secret for an application
+  """
+  regenerateOAuthAPIProviderSecret(
+    input: RegenerateOAuthAPIProviderSecretInput!
+  ): RegenerateOAuthAPIProviderSecretPayload!
+
+  """
+  Create an OAuth 2.0 authorization code
+  """
+  createOAuthAPICode(input: CreateOAuthAPICodeInput!): CreateOAuthAPICodePayload!
+}
+
+extend type Query {
+  """
+  Get an OAuth 2.0 Application by ID
+  """
+  oauthAPIProvider(id: ID!): OAuthAPIProvider
```

Comment:
The resolve function for this looks like it requires an authToken. Therefore, this belongs on the `User` object, not the `Query` object.